src/main/kotlin/com/nikitanazarov/coroutinestacks/ui/CoroutineStacksPanel.kt

```diff
@@ -198,25 +311,81 @@ class CoroutineStacksPanel(project: Project) : JBPanelWithEmptyText() {
     }
 
     inner class LibraryFrameToggle(
-        suspendContextImpl: SuspendContextImpl,
-        dispatcherToCoroutineDataList: Map<String, List<CoroutineInfoData>>
-    ) : JToggleButton(AllIcons.General.Filter) {
+        private val suspendContextImpl: SuspendContextImpl,
+        private val dispatcherToCoroutineDataList: Map<String, List<CoroutineInfoData>>,
+        private val dispatcherDropdownMenu: DispatcherDropdownMenu
+    ) : JButton(AllIcons.General.Filter) {
         init {
-            text = CoroutineStacksBundle.message("hide.library.frames")
+            transparent = true
+            setToolTip()
+            addActionListener()
+        }
+
+        private fun setToolTip() {
+            toolTipText = if (areLibraryFramesAllowed) {
+                CoroutineStacksBundle.message("hide.library.frames")
+            } else {
+                CoroutineStacksBundle.message("show.library.frames")
+            }
+        }
+
+        private fun addActionListener() {
             addActionListener {
-                CoroutineStackPanelData.apply {
-                    areLibraryFramesAllowed = areLibraryFramesAllowed.not()
-                    text = if (areLibraryFramesAllowed)
-                        CoroutineStacksBundle.message("hide.library.frames")
-                    else
-                        CoroutineStacksBundle.message("show.library.frames")
-                }
-                val selectedDispatcher = dispatcherDropdownMenu?.selectedItem as? String
+                areLibraryFramesAllowed = areLibraryFramesAllowed.not()
+                transparent = areLibraryFramesAllowed
+                setToolTip()
+
+                val selectedDispatcher = dispatcherDropdownMenu.selectedItem as? String
                 val coroutineDataList = dispatcherToCoroutineDataList[selectedDispatcher]
                 if (!coroutineDataList.isNullOrEmpty()) {
                     updateCoroutineStackForest(coroutineDataList, suspendContextImpl)
                 }
             }
         }
     }
+
+    private fun JBScrollPane.scale(scaleFactor: Float) {
+        val viewport = viewport
+        val view = viewport.view as? Container ?: return
+
+        view.components.forEach { component ->
+            if (component is JBList<*>) {
+                with(component) {
+                    font = font.deriveFont(font.size2D + initialFontSize * scaleFactor)
+                    preferredWidth += (initialWidth * scaleFactor).toInt()
+                    fixedCellHeight += (initialCellHeight * scaleFactor).toInt()
+                    preferredHeight = fixedCellHeight * model.size
+                }
+            }
+        }
+    }
 }
+
+object OriginalListState {
+    const val initialFontSize = 12f
```

Comment:
How did you come up with these numbers? Unfortunately if the internals of JBList will change some day, the code that depends on this numbers will not work correctly. Instead, we should fetch initial sizes once we build the coroutine stacks panel for the first time!